trusted_applet/internal/storage/persistence.go

```diff
@@ -101,6 +110,37 @@ func (p *SlotPersistence) Logs(_ context.Context) ([]string, error) {
 	return r, nil
 }
 
+// marshalCheckpoint knows how to serialise a checkpoint for storage by the
+// persistence.
+func marshalCheckpoint(cpRaw []byte) []byte {
+	// Return a new slice which contains the magic raw prefix followed directly by the checkpoint bytes.
+	return append(append([]byte{}, rawRecordMagic...), cpRaw...)
```

Comment:
Maybe the compiler is smart enough, but this looks like it could allocate a bunch of slices here. It's not as elegant, but I'd be interested to know if there's a performance difference with an explicit allocation:

```
result := make([]byte, 0, len(rawRecordMagic)+len(cpRaw))
result = append(result, rawRecordMagic...)
return append(result, cpRaw...)
```

Maybe you already checked this though?